### PR TITLE
nextcloud-talk-desktop: init at 1.0.0

### LINKED ISSUE #253243: Package request: Nextcloud Talk Desktop
**Project description**
Nextcloud Talk Desktop Client

**Metadata**

* homepage URL: https://nextcloud.com/talk/
* source URL: https://github.com/nextcloud/talk-desktop
* license: AGPL-3.0 license 
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

closes https://github.com/NixOS/nixpkgs/issues/253243

https://nextcloud.com/talk/
https://github.com/nextcloud/talk-desktop

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 302781` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nextcloud-talk</li>
  </ul>
</details>

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 302781` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nextcloud-talk</li>
  </ul>
</details>

**Comment by prusnak** [comment]
I think the package should be named `nextcloud-talk-desktop` to make it obvious this ist the client part, not the server.

**Comment by Thatoo** [comment]
any news on this package?

File: pkgs/by-name/ne/nextcloud-talk-desktop/package.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    mkdir -p $out/opt
+
+    cp -r $src/* $out/opt/
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Link the application in $out/bin away from contents of `preInstall`
+    ln -s "$out/opt/Nextcloud Talk" $out/bin/${finalAttrs.pname}
```
Comment (FliegendeWurst) [inline pkgs/by-name/ne/nextcloud-talk-desktop/package.nix:76]:
```suggestion
    ln -s "$out/opt/Nextcloud Talk" $out/bin/nextcloud-talk-desktop
```
#277994

File: pkgs/by-name/ne/nextcloud-talk-desktop/package.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Nextcloud Talk Desktop Client Preview";
+    homepage = "https://github.com/nextcloud/talk-desktop";
+    changelog = "https://github.com/nextcloud/talk-desktop/blob/${finalAttrs.version}/CHANGELOG.md";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ kashw2 ];
+    mainProgram = finalAttrs.pname;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ne/nextcloud-talk-desktop/package.nix:87]:
```suggestion
    mainProgram = finalAttrs.pname;
    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
```

Comment (SuperSandro2000) [inline pkgs/by-name/ne/nextcloud-talk-desktop/package.nix:87]:
```suggestion
    mainProgram = "nextcloud-talk-desktop";
```
the entire point of introducing mainProgram was to no longer do that

**Comment by CaptainJawZ** [comment]
@_@ pardon the intrusion with the suggested changes, not sure if it's appropriate to upgrade apps that have not been merged yet, but since the official 1.0.0 release came out, I figured this could be bumped to that instead.

File: pkgs/by-name/ne/nextcloud-talk-desktop/package.nix [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Nextcloud Talk Desktop Client";
+    homepage = "https://github.com/nextcloud/talk-desktop";
+    changelog = "https://github.com/nextcloud/talk-desktop/blob/${finalAttrs.version}/CHANGELOG.md";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ kashw2 ];
+    mainProgram = "nextcloud-talk-desktop";
+    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ne/nextcloud-talk-desktop/package.nix:87]:
```suggestion
    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
    platforms = [ "x86_64-linux" ];
```

File: pkgs/by-name/ne/nextcloud-talk-desktop/package.nix [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    mkdir -p $out/opt
+
+    cp -r $src/* $out/opt/
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Link the application in $out/bin away from contents of `preInstall`
+    ln -s "$out/opt/Nextcloud Talk-linux-x64" $out/bin/nextcloud-talk-desktop
```
Comment (FliegendeWurst) [inline pkgs/by-name/ne/nextcloud-talk-desktop/package.nix:75]:
```suggestion
    ln -s "$out/opt/Nextcloud Talk-linux-x64/Nextcloud Talk" $out/bin/nextcloud-talk-desktop
```

Comment (kashw2) [reply pkgs/by-name/ne/nextcloud-talk-desktop/package.nix:75]: Thanks for the catch

**Comment by Thatoo** [comment]
Great news.Woud it be possible to backport it to 24.11

**Comment by Thatoo** [comment]
I have tested it on NixOS 24.11 Cosmic and unfortunately, this app doesn't appear in the app launcher as if no .desktop file could be find by the system.
Any idea why?

**Comment by FliegendeWurst** [comment]
That is because there was no .desktop file. Check #367079

**Pull request merged**